reader.go

```diff
@@ -721,6 +723,68 @@ func decodeLineOfMediaPlaylist(p *MediaPlaylist, wv *WV, state *decodingState, l
 		state.scte = new(SCTE)
 		state.scte.Syntax = SCTE35_OATCLS
 		state.scte.CueType = SCTE35Cue_End
+	case !state.tagSCTE35 && strings.HasPrefix(line, "#EXT-X-DATERANGE:"):
+		tagSCTE35 := false
+		scte := new(SCTE)
+		scte.Syntax = SCTE35_DATERANGE
+		for attribute, value := range decodeParamsLine(line[12:]) {
+			switch attribute {
+			case "SCTE35-CMD":
+				tagSCTE35 = true
+				buf, err := hex.DecodeString(strings.TrimLeft(value, "0x"))
+				if err != nil {
+					return err
+				}
+				scte.Cue = base64.StdEncoding.EncodeToString(buf)
+				scte.CueType = SCTE35Cue_Cmd
+			case "ID":
+				scte.ID = value
+			case "START-DATE":
+				startDate, err := time.Parse(DATETIME, value)
+				if err != nil {
+					return err
+				}
+				scte.StartDate = &startDate
+			case "END-DATE":
+				endDate, err := time.Parse(DATETIME, value)
+				if err != nil {
+					return err
+				}
+				scte.EndDate = &endDate
+			case "DURATION":
+				duration, err := strconv.ParseFloat(value, 64)
+				if err != nil {
+					return err
+				}
+				scte.Duration = &duration
+			case "PLANNED-DURATION":
+				plannedDuration, err := strconv.ParseFloat(value, 64)
+				if err != nil {
+					return err
+				}
+				scte.PlannedDuration = &plannedDuration
+			case "SCTE35-OUT":
+				tagSCTE35 = true
+				scte.CueType = SCTE35Cue_Start
+				buf, err := hex.DecodeString(strings.TrimLeft(value, "0x"))
+				if err != nil {
+					return err
+				}
+				scte.Cue = base64.StdEncoding.EncodeToString(buf)
+			case "SCTE35-IN":
+				scte.CueType = SCTE35Cue_End
+				buf, err := hex.DecodeString(strings.TrimLeft(value, "0x"))
+				if err != nil {
+					return err
+				}
+				tagSCTE35 = true
```

Comment:
Why for OUT this is assigned first but for IN it is assigned last?

Reply:
It doesn't matter, I just put the line randomly